File: src/main/java/cn/labzen/logger/reload4j/LabzenReload4jServiceProvider.java
package cn.labzen.logger.reload4j;

import org.slf4j.ILoggerFactory;
import org.slf4j.reload4j.Reload4jServiceProvider;

public class LabzenReload4jServiceProvider extends Reload4jServiceProvider {

  private ILoggerFactory loggerFactory;

  @Override
  public void initialize() {
    super.initialize();

    ILoggerFactory principalFactory = super.getLoggerFactory();
    loggerFactory = new LabzenReload4jLoggerFactory(principalFactory);
  }

  @Override
  public ILoggerFactory getLoggerFactory() {
    return loggerFactory;
  }
}

File: src/main/kotlin/cn/labzen/logger/reload4j/LabzenReload4jServiceProvider.kt
package cn.labzen.logger.reload4j

import org.slf4j.ILoggerFactory
import org.slf4j.reload4j.Reload4jServiceProvider

class LabzenReload4jServiceProvider : Reload4jServiceProvider() {

  private lateinit var labzenLoggerFactory: ILoggerFactory

  override fun initialize() {
    super.initialize()

    val principalFactory = super.getLoggerFactory()
    labzenLoggerFactory = LabzenReload4jLoggerFactory(principalFactory)
  }

  override fun getLoggerFactory(): ILoggerFactory =
    labzenLoggerFactory
}
